helix-tui/src/backend/crossterm.rs

```diff
@@ -23,13 +23,32 @@ use std::{
     fmt,
     io::{self, Write},
 };
+use termini::TermInfo;
 
 fn term_program() -> Option<String> {
-    std::env::var("TERM_PROGRAM").ok()
+    // Some terminals don't set $TERM_PROGRAM
+    match std::env::var("TERM_PROGRAM") {
+        Err(_) => std::env::var("TERM").ok(),
+        Ok(term_program) => Some(term_program),
+    }
 }
 fn vte_version() -> Option<usize> {
     std::env::var("VTE_VERSION").ok()?.parse().ok()
 }
+fn reset_cursor_approach(terminfo: TermInfo) -> String {
+    let Some(termini::Value::Utf8String(se_str)) = terminfo.extended_cap("Se") else {
+        return "".to_string();
+    };
```

Comment:
If there's no `Se` capability we don't end up emitting the fallback? I thought the intent was to always use the fallback.

Reply:
this current implementation always emits the fallback, (`Se` or an empty string), and `cnorm` in that order

That `let se_str` is written that way because I couldn't figure out a cleaner way to access the value in `Option<Value<'_>>` that `.extended_cap("Se")` returns.

Reply:
No, this snippet early returns and an empty string is emitted.

Reply:
oh, I see, thank you!!